src/backend/src/services/onboarding.services.ts

```diff
@@ -319,4 +318,81 @@ export default class OnboardingServices {
       data: { dateDeleted: new Date(), userDeletedId: deleter.userId }
     });
   }
+
+  static async toggleChecklistItem(checklistId: string, userId: string) {
+    const checklist = await prisma.checklist.findUnique({
+      where: { checklistId },
+      include: { usersChecked: true, subtasks: true }
+    });
+
+    if (!checklist) {
+      throw new NotFoundException('Checklist', checklistId);
+    }
+
+    const isChecked = checklist.usersChecked.some((user) => user.userId === userId);
+
+    if (isChecked) {
+      await prisma.checklist.update({
+        where: { checklistId },
+        data: {
+          usersChecked: {
+            disconnect: { userId }
+          }
+        }
+      });
+    } else {
+      await prisma.checklist.update({
+        where: { checklistId },
+        data: {
+          usersChecked: {
+            connect: { userId }
+          }
+        }
+      });
+    }
+
+    if (!checklist.parentChecklistId) {
+      return checklist;
+    }
+
+    const parentChecklist = await prisma.checklist.findUnique({
+      where: { checklistId: checklist.parentChecklistId },
+      include: {
+        subtasks: {
+          where: { dateDeleted: null },
+          include: { usersChecked: true }
+        }
+      }
+    });
+
+    if (!parentChecklist) {
+      return checklist;
+    }
```

Comment:
Instead of returning if there is no parentChecklist, can you just put all the code under this in a if block. so like `if (parentChecklist) { the code for the updating }`